### PR TITLE
ABAP File Format Version 2 for CFDF - Custom Field

### PR DESCRIPTION
**Pull request by ReneDehn**
New Version for AFF Type Custom Field

### CODE REVIEW

**Comment by albertmink** [comment]
@ReneDehn don't get confused by the false positive in `Validation JSON examples`. Need a fix from us (tracked here #502) and if this is blocking, we might overrule the check.

**Comment by ReneDehn** [comment]
@albertmink Thanks for the info. I just fixed the issue regarding abaplint.
So, if this is a false positive in Validation JSON examples we can wait for your fix or you can overrule the check. Thanks.

**Comment by albertmink** [comment]
> @albertmink Thanks for the info. I just fixed the issue regarding abaplint. So, if this is a false positive in Validation JSON examples we can wait for your fix or you can overrule the check. Thanks.

Solved in https://github.com/SAP/abap-file-formats/commit/6a8246a2db0a46de5b5ff6d97b01ca109b462a49 😸

File: file-formats/cfdf/type/zif_aff_cfdf_v2.intf.abap [hunk 1]
```diff
@@ -0,0 +1,573 @@
...
+  TYPES ty_aggregation            TYPE c LENGTH 5.
+  TYPES ty_component_kind         TYPE c LENGTH 11.
+  TYPES:
+    "! <p class="shorttext">Labels and Tooltip</p>
+    "! Labels and tooltip
+    BEGIN OF ty_gs_ui_text,
+      "! <p class="shorttext">Translation Language</p>
+      "! Translation language
+      "! $required
+      language             TYPE sy-langu,
```
Comment (schneidermic0) [inline file-formats/cfdf/type/zif_aff_cfdf_v2.intf.abap:21]: `language` has been discussed earlier:
https://github.com/SAP/abap-file-formats/pull/401#discussion_r984269158

**Review by schneidermic0 — COMMENTED** [review]
Thanks, @ReneDehn.

File: file-formats/cfdf/type/zif_aff_cfdf_v2.intf.abap [hunk 2]
```diff
@@ -0,0 +1,573 @@
...
+      "! <p class="shorttext">Field Medium Label</p>
+      "! Field medium label
+      field_medium_label   TYPE c                 LENGTH 20,
+
+      "! <p class="shorttext">Field Short Label</p>
+      "! Field short label
+      field_short_label    TYPE c                 LENGTH 10,
+    END OF ty_gs_ui_text.
+  TYPES ty_gt_ui_text TYPE STANDARD TABLE OF ty_gs_ui_text
+        WITH KEY language.
```
[the remark this answers is not in the document]
Comment (ReneDehn) [reply file-formats/cfdf/type/zif_aff_cfdf_v2.intf.abap:58]: component_kind is already set to $required in the AFF V2 interface.

But you are right, I forgot to set them as key fields in the table type. I will correct this.

File: file-formats/cfdf/README.md [hunk 3]
```diff
@@ -3,3 +3,4 @@
 File | Cardinality | Definition | Schema | Example
```
[the remark this answers is not in the document]
Comment (ReneDehn) [reply file-formats/cfdf/README.md:3]: That's fine for me

[the remark this answers is not in the document]
Comment (ReneDehn) [reply file-formats/cfdf/README.md:3]: @schneidermic0 I fixed the README.md. I hope this is what you had in mind.

File: file-formats/cfdf/type/zif_aff_cfdf_v2.intf.abap (hunk 2 above)
[the remark this answers is not in the document]
Comment (ReneDehn) [reply file-formats/cfdf/type/zif_aff_cfdf_v2.intf.abap]: @schneidermic0 key fields of the table type are now complete. Thanks for pointing this out.

File: file-formats/cfdf/type/zif_aff_cfdf_v2.intf.abap [hunk 4]
```diff
@@ -0,0 +1,573 @@
...
+      "! <p class="shorttext">Field Medium Label</p>
+      "! Field medium label
+      field_medium_label   TYPE c                 LENGTH 20,
+
+      "! <p class="shorttext">Field Short Label</p>
+      "! Field short label
+      field_short_label    TYPE c                 LENGTH 10,
+    END OF ty_gs_ui_text.
+  TYPES ty_gt_ui_text TYPE STANDARD TABLE OF ty_gs_ui_text
+        WITH KEY language component_kind component_identifier.
```
Comment (schneidermic0) [inline file-formats/cfdf/type/zif_aff_cfdf_v2.intf.abap:58]: Isn't language missing in the key, now?

File: file-formats/cfdf/type/zif_aff_cfdf_v2.intf.abap (hunk 2 above)
[the remark this answers is not in the document]
Comment (schneidermic0) [reply file-formats/cfdf/type/zif_aff_cfdf_v2.intf.abap]: See https://github.com/SAP/abap-file-formats/pull/501#discussion_r1126709532

File: file-formats/cfdf/type/zif_aff_cfdf_v2.intf.abap (hunk 4 above)
Comment (ReneDehn) [reply file-formats/cfdf/type/zif_aff_cfdf_v2.intf.abap:58]: > Isn't `language` missing in the key, now?

What do you mean? language is part of the key:
  TYPES ty_gt_ui_text TYPE STANDARD TABLE OF ty_gs_ui_text
        WITH KEY language component_kind component_identifier.

Comment (schneidermic0) [reply file-formats/cfdf/type/zif_aff_cfdf_v2.intf.abap:58]: Sorry, my fault. I just somehow missed it because `language` is rendered as keyword.

**Review by schneidermic0 — APPROVED** [review]
File format looks good to me

**Pull request merged**